### ISSUE TITLE
Data channel testing

### ISSUE THREAD

**Issue by lgrahl** [webrtc/KITE#9]
Hey everyone. Thanks for the presentation earlier. Sadly, the sound was cut off a lot, so it was very hard to follow you.

As part of my master thesis, I'm going to test (SCTP-based) data channel implementations. However, only a fraction of those tests will be made for the WebRTC JS API. Some things aren't testable on that level, for example *abandoning* a partially sent message on an unreliable channel. This will require being able to inject packets at SCTP level (so, on top of ICE/DTLS).

My idea is to use [RAWRTC](https://github.com/rawrtc/rawrtc) for that purpose as it allows to inject arbitrary packets on top of ICE/DTLS easily and can talk to any other WebRTC implementation. In combination with the [SCTP test tool (stt)](https://github.com/nplab/stt), this would allow to inject a sequence of SCTP packets and make those tests possible.

My main questions are the following:
1. Is this something that's possible to do with KITE's architecture?
2. I would also like to add performance tests (including tests with delay, etc.) but I'm unsure whether this fits into KITE's architecture.
3. What components would I have to look at to get started?

**Comment by agouaillard-cosmo** [comment]
Hi lennart.

Thank you for your hard work on data channel, and the recent improvement it triggered in several browser, including Firefox.

KITE is just an engine to distribute arbitrary tests across a large heterogeneous grid of *CLIENTS*. The Open source version is going to focus on Browsers, with desktop browsers support already in, and mobile browsers support coming soon, all thanks to google support.

At CoSMo we are developing some extra modules for Native clients, including Qt-based clients, and electron. Using rawRTC would fit in that bucket. It requires writing an additional Selenium/webdriver compliant layer so that KITE can instrument it. Since you already support cmake (kudos), it wouldn't be too difficult to include your code in a Qt client and then test it with kite against browser vendors. Another way is to prepare a modify (weaponized) versions of libwertc with your SCTP specific modifications and put it in a modified version of chrome.

I think you are a master student right now, if you supervisor is interested, we can collaborate on this a write a small conference paper out of it. We do have funding for R&D collaborations, including funding for Master Thesis, and are already hosting 10 master Thesis between February and October  2018.

**Comment by agouaillard-cosmo** [comment]
No activity from the original author. closing.
